microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -643,5 +789,7 @@ func (r *GMConnectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&mcv1alpha3.GMConnector{}).
 		WithEventFilter(ignoreStatusUpdatePredicate). // Use the predicate here
+		Owns(&appsv1.Deployment{}).
```

Comment:
suppose we can also get service and deployment update and we can Update Status in annotation as well